docker/Dockerfile

```diff
@@ -4,11 +4,13 @@ ENV DEBIAN_FRONTEND noninteractive
 RUN apt-get update && apt-get install --yes curl wget zip git build-essential pkg-config libcjson-dev openjdk-17-jdk libbcprov-java libgoogle-gson-java libguava-java libssl-dev gcc g++ libbotan-2-dev nlohmann-json3-dev python3 python3-pip rustc cargo iputils-ping telnet wget && rm -fr /var/cache/apt/* /var/lib/apt/lists/*
 RUN pip3 install --break-system-packages cryptography requests
 
+# Install latest Go environment
 RUN GO_VERSION=$(wget -qO- https://go.dev/VERSION?m=text | head -n 1) && wget https://go.dev/dl/${GO_VERSION}.linux-amd64.tar.gz && rm -rf /usr/local/go && tar -C /usr/local -xzf ${GO_VERSION}.linux-amd64.tar.gz && rm -rf ${GO_VERSION}.linux-amd64.tar.gz
 ENV PATH $PATH:/usr/local/go/bin
 
-COPY rust /rust
-RUN cd /rust && cargo vendor
+# Install Rust and vendor in dependencies
+COPY rust /tmp/rust
+RUN cd /tmp/rust && cargo vendor && rm -fr /tmp/rust
```

Comment:
This won't work. cargo vendor downloads the source codes of the specified crates into the cwd, but you are deleting it immediately afterward. Removing `/tmp/rust` undos the vendoring.

Reply:
To clarify: Currently, our projects compile all the dependencies. In the current configuration, no caching happens before our executables are compiled.